Dev

MarselMingaripov opened the pull request.

MarselMingaripov committed:
create features branches

vladimirivn committed:
Создание ветки - vladimirivnFeature

vladimirivn committed:
Создание ветки - vladimirivnFeature

vladimirivn committed:
Merge pull request #2 from MarselMingaripov/vladimirivnFeature

Vladimirivn feature

elenamatvienko committed:
commit

MarselMingaripov committed:
create features branches

MarselMingaripov committed:
create project structure

MarselMingaripov committed:
create project structure

MarselMingaripov committed:
create project structure

MarselMingaripov committed:
add readme

MarselMingaripov committed:
add user, trialPeriod, report models

vladimirivn committed:
Коммит

MarselMingaripov committed:
add readme

MarselMingaripov committed:
add readme

vladimirivn committed:
Merge branch 'dev' into vladimirivnFeature-1

vladimirivn committed:
Коммит

elenamatvienko committed:
Написала класс животный

elenamatvienko committed:
Merge branch 'dev' into elenaFeature1

MarselMingaripov committed:
add owners, repo

vladimirivn committed:
Создал базовый класс AnimalShelter,подклассы CatsShelter и DogsShelter.

vladimirivn committed:
Создал базовый класс AnimalShelter,подклассы CatsShelter и DogsShelter.

elenamatvienko committed:
Внесла изменения в класс Животные

elenamatvienko committed:
Merge branch 'elenaFeature1' into dev

MarselMingaripov committed:
add catOwnerService

MarselMingaripov committed:
add trialPeriod, user services

elenamatvienko committed:
Написала сервисы для MessageToVolunteer & Report

MarselMingaripov committed:
Merge remote-tracking branch 'origin/dev' into dev

SergeyGolybtsov committed:
Создал репозиторий и сервисы.

MarselMingaripov committed:
add info methods in dogShelterService

vladimirivn committed:
Создал перечисления с типами правил и рекомендаций для потенциальных хозяев животных от каждого приюта.

vladimirivn committed:
Merge remote-tracking branch 'origin/dev' into dev

MarselMingaripov committed:
Merge branch 'master' into dev

vladimirivn committed:
Убрано поле из базового класса AnimalShelter. В классы CatShelter, DogShelter добавлен конструктор с базовыми полями. Добавлен в DogRepository метод findByName. Добавлены и реализованы методы findByName, addDogToShelter в сервисы DogService и DogShelterService.

vladimirivn committed:
Merge branch 'vladimirivnFeature-1' into dev

# Conflicts:
#	src/main/java/com/example/petshelter/service/DogShelterService.java
#	src/main/java/com/example/petshelter/service/impl/DogShelterServiceImpl.java

vladimirivn committed:
Добавлен и реализован метод addDogToShelter в сервис DogShelterService.

elenamatvienko committed:
Выполнила задачи: сервис DogOwner, добавила информационные методы из 1 этапа ТЗ в CatShelter, метод добавления Cat. Методы добавления CatOwner и DogOwner нужно проверить

MarselMingaripov committed:
add migrations

MarselMingaripov committed:
Merge remote-tracking branch 'origin/dev' into dev

# Conflicts:
#	src/main/java/com/example/petshelter/entity/shelter/СatShelter.java

vladimirivn committed:
Добавил ValidationService

MarselMingaripov committed:
add migrations

MarselMingaripov committed:
поправил метод добавления хозяев котов в кошачий приют

MarselMingaripov committed:
удалил животное

vladimirivn committed:
Merge branch 'dev' into vladimirivnFeature-1

# Conflicts:
#	src/main/java/com/example/petshelter/entity/shelter/CatShelter.java

vladimirivn committed:
Изменения

MarselMingaripov committed:
поправил catShelter

MarselMingaripov committed:
добавил user, report, trialPeriod, message контроллеры

elenamatvienko committed:
Доработала метод добавления хозяев собак, реализовала контроллеры Cat, Dog, CatOwner, DogOwner

elenamatvienko committed:
Merge remote-tracking branch 'origin/dev' into dev

vladimirivn committed:
Создан сервис валидации. Добавлена проверка при создании сущностей в методы

vladimirivn committed:
Создан сервис валидации. Добавлена проверка при создании сущностей в методы

vladimirivn committed:
Merge remote-tracking branch 'origin/dev' into dev

# Conflicts:
#	src/main/java/com/example/petshelter/service/impl/DogOwnerServiceImpl.java
#	src/main/java/com/example/petshelter/service/impl/DogServiceImpl.java
#	src/main/java/com/example/petshelter/service/impl/DogShelterServiceImpl.java

vladimirivn committed:
Добавил изменение в DogServiceImpl

MarselMingaripov committed:
добавил user, report, trialPeriod, message контроллеры

elenamatvienko committed:
Доработала контроллеры Cat, Dog, CatOwner, DogOwner. Реализовала контроллеры CatShelter, DogShelter. Добавила описание API.

MarselMingaripov committed:
добавил обработчик ошибок, переделал миграции

MarselMingaripov committed:
Merge branch 'dev' into marselFeature-1

vladimirivn committed:
Добавил описания в классы JavaDoc

vladimirivn committed:
Merge branch 'vladimirivnFeature-1' into dev

MarselMingaripov committed:
Merge branch 'dev' into marselFeature-1

MarselMingaripov committed:
поправил кошачий контроллер

vladimirivn committed:
Добавил описания JavaDoc в сервисы CatService, CatServiceImpl, CatOwnerService

vladimirivn committed:
Merge branch 'vladimirivnFeature-1' into dev

vladimirivn committed:
Добавил и реализовал в сервис ValidationService проверку номера телефона по шаблону. Внес изменения в методы ValidationServiceImpl и UserServiceImpl

vladimirivn committed:
Merge branch 'vladimirivnFeature-1' into dev

MarselMingaripov committed:
добавил бронь котов

dmitry-bizin commented on pom.xml:
```diff
@@ -43,6 +43,48 @@
+        <dependency>
+            <groupId>org.liquibase</groupId>
+            <artifactId>liquibase-core</artifactId>
+            <version>4.20.0</version>
```
версию писать не нужно, чтобы избежать ада конфликта версий

dmitry-bizin commented on pom.xml:
```diff
@@ -43,6 +43,48 @@
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-data-jpa</artifactId>
+            <version>2.7.7</version>
```
версию писать не нужно, чтобы избежать ада конфликта версий

dmitry-bizin commented on pom.xml:
```diff
@@ -43,6 +43,48 @@
+        <!-- https://mvnrepository.com/artifact/org.hibernate.orm/hibernate-community-dialects -->
+        <dependency>
+            <groupId>org.hibernate.orm</groupId>
+            <artifactId>hibernate-community-dialects</artifactId>
```
эта зависимость не нужна

dmitry-bizin commented on pom.xml:
```diff
@@ -43,6 +43,48 @@
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-validation</artifactId>
+            <version>2.7.7</version>
```
версию писать не нужно, чтобы избежать ада конфликта версий

dmitry-bizin commented on pom.xml:
```diff
@@ -43,6 +43,48 @@
+        <!-- https://mvnrepository.com/artifact/org.yaml/snakeyaml -->
+        <dependency>
+            <groupId>org.yaml</groupId>
+            <artifactId>snakeyaml</artifactId>
```
эта зависимость не нужна

dmitry-bizin commented on src/main/java/com/example/petshelter/controller/CatController.java:
```diff
@@ -0,0 +1,76 @@
+    @ApiResponse(responseCode = "200", description = " Запрос выполнен, данные получены")
+    @ApiResponse(responseCode = "404", description = "Параметры запроса отсутствуют или имеют некорректный формат")
+    @ApiResponse(responseCode = "500", description = "Произошла ошибка, не зависящая от вызывающей стороны")
+    public ResponseEntity<Cat> findByName(@PathVariable String name) {
```
имя лучше сделать параметром запроса, а не пути. В пути можно передавать только id (по REST)

dmitry-bizin commented on src/main/java/com/example/petshelter/controller/CatOwnerController.java:
```diff
@@ -0,0 +1,77 @@
+        return ResponseEntity.ok().body(catOwnerService.findById(id));
+    }
+
+    @GetMapping("/find-by-phone-number/{phoneNumber}")
```
телефон лучше сделать параметром запроса, а не пути. В пути можно передавать только id (по REST)

SergeyGolybtsov committed:
Создал тесты.

dmitry-bizin commented on src/main/java/com/example/petshelter/controller/DogController.java:
```diff
@@ -0,0 +1,77 @@
+        return ResponseEntity.ok().body(dogService.findById(id));
+    }
+
+    @GetMapping("/find-by-name/{name}")
```
имя лучше сделать параметром запроса, а не пути. В пути можно передавать только id (по REST)

dmitry-bizin commented on src/main/java/com/example/petshelter/controller/DogOwnerController.java:
```diff
@@ -0,0 +1,78 @@
+        return ResponseEntity.ok().body(dogOwnerService.findById(id));
+    }
+
+    @GetMapping("/find-by-phone-number/{phoneNumber}")
```
телефон лучше сделать параметром запроса, а не пути. В пути можно передавать только id (по REST)

dmitry-bizin commented on src/main/java/com/example/petshelter/entity/Cat.java:
```diff
@@ -0,0 +1,53 @@
+    /**
+     * Имя животного
+     */
+    String name;
```
все поля должны быть приватные

BizinMitya commented on src/main/java/com/example/petshelter/entity/Cat.java:
```diff
@@ -0,0 +1,53 @@
+    /**
+     * Статус нахождения животного в приюте
+     */
+    private StatusAnimal statusAnimal;
```
здесь должна быть аннотация Enumerated со значением String

BizinMitya commented on src/main/java/com/example/petshelter/entity/Dog.java:
```diff
@@ -0,0 +1,44 @@
+    /**
+     * Статус нахождения животного в приюте
+     */
+    private StatusAnimal statusAnimal;
```
здесь должна быть аннотация Enumerated со значением STRING

dmitry-bizin commented on src/main/java/com/example/petshelter/entity/Dog.java:
```diff
@@ -0,0 +1,44 @@
+    /**
+     * Имя животного
+     */
+    String name;
```
все поля должны быть приватные

dmitry-bizin commented on src/main/java/com/example/petshelter/entity/TrialPeriod.java:
```diff
@@ -0,0 +1,51 @@
+    private List<Report> reports;
+    /**
+     * Результат по прохождению испытательного срока опекуном животного
+     */
+    private TrialPeriodResult result;
```
здесь должна быть аннотация Enumerated со значением STRING

BizinMitya commented on src/main/java/com/example/petshelter/entity/TrialPeriod.java:
```diff
@@ -0,0 +1,51 @@
+    /**
+     * Отчеты опекунов животного
+     */
+    private List<Report> reports;
```
где аннотация связи?

dmitry-bizin commented on src/main/java/com/example/petshelter/repository/CatOwnerRepository.java:
```diff
@@ -0,0 +1,16 @@
+@EnableJpaRepositories
+public interface CatOwnerRepository extends JpaRepository<CatOwner, Long> {
+
+    Boolean existsByPhoneNumber(String phoneNumber);
```
лучше вернуть boolean, а не Boolean

dmitry-bizin commented on src/main/java/com/example/petshelter/repository/DogOwnerRepository.java:
```diff
@@ -0,0 +1,12 @@
+import java.util.Optional;
+
+public interface DogOwnerRepository extends JpaRepository<DogOwner, Long> {
+    Boolean existsByPhoneNumber(String phoneNumber);
```
лучше вернуть boolean, а не Boolean

dmitry-bizin commented on src/main/java/com/example/petshelter/repository/UserRepository.java:
```diff
@@ -0,0 +1,11 @@
+@Repository
+public interface UserRepository extends JpaRepository<User, Long> {
+
+    Boolean existsByPhoneNumber(String phoneNumber);
```
лучше вернуть boolean, а не Boolean

dmitry-bizin commented on src/main/java/com/example/petshelter/service/impl/CatOwnerServiceImpl.java:
```diff
@@ -0,0 +1,76 @@
+    @Override
+    public CatOwner findById(Long id) {
+        if (catOwnerRepository.findById(id).isPresent()) {
+            return catOwnerRepository.findById(id).get();
```
Зачем 2 раза запрос в БД делать? Надо сделать 1 запрос в БД, получить результат и далее работать с результатом. У вас приложение просто в 2 раза тормозит на пустом месте.

dmitry-bizin commented on src/main/java/com/example/petshelter/service/impl/CatOwnerServiceImpl.java:
```diff
@@ -0,0 +1,76 @@
+    public CatOwner deleteById(Long id) {
+        if (catOwnerRepository.findById(id).isPresent()) {
+            catOwnerRepository.deleteById(id);
+            return catOwnerRepository.findById(id).get();
```
Зачем 2 раза запрос в БД делать? Надо сделать 1 запрос в БД, получить результат и далее работать с результатом. У вас приложение просто в 2 раза тормозит на пустом месте.

dmitry-bizin commented on src/main/java/com/example/petshelter/service/impl/CatServiceImpl.java:
```diff
@@ -0,0 +1,113 @@
+    @Override
+    public Cat findById(Long id) {
+        if (catRepository.findById(id).isPresent()) {
+            return catRepository.findById(id).get();
```
Зачем 2 раза запрос в БД делать? Надо сделать 1 запрос в БД, получить результат и далее работать с результатом. У вас приложение просто в 2 раза тормозит на пустом месте.

dmitry-bizin commented on src/main/java/com/example/petshelter/service/impl/CatServiceImpl.java:
```diff
@@ -0,0 +1,113 @@
+    @Override
+    public Cat deleteById(Long id) {
+        if (catRepository.findById(id).isPresent()){
+            Cat cat = findById(id);
```
Зачем 2 раза запрос в БД делать? Надо сделать 1 запрос в БД, получить результат и далее работать с результатом. У вас приложение просто в 2 раза тормозит на пустом месте.

dmitry-bizin commented on src/main/java/com/example/petshelter/service/impl/CatShelterServiceImpl.java:
```diff
@@ -0,0 +1,102 @@
+    @Override
+    public CatShelter findById(Long id) {
+        if (catShelterRepository.findById(id).isPresent()) {
+            return catShelterRepository.findById(id).get();
```
Зачем 2 раза запрос в БД делать? Надо сделать 1 запрос в БД, получить результат и далее работать с результатом. У вас приложение просто в 2 раза тормозит на пустом месте.

dmitry-bizin commented on src/main/java/com/example/petshelter/service/impl/CatShelterServiceImpl.java:
```diff
@@ -0,0 +1,102 @@
+    public CatShelter deleteById(Long id) {
+        if (catShelterRepository.findById(id).isPresent()) {
+            catShelterRepository.deleteById(id);
+            return catShelterRepository.findById(id).get();
```
Зачем 2 раза запрос в БД делать? Надо сделать 1 запрос в БД, получить результат и далее работать с результатом. У вас приложение просто в 2 раза тормозит на пустом месте.

dmitry-bizin commented on src/main/java/com/example/petshelter/service/impl/CatShelterServiceImpl.java:
```diff
@@ -0,0 +1,102 @@
+
+    @Override
+    public String returnInformation() {
+        return findById(1L).getInformation();
```
что за магические числа? почему не 100?

dmitry-bizin commented on src/main/java/com/example/petshelter/service/impl/CatShelterServiceImpl.java:
```diff
@@ -0,0 +1,102 @@
+
+    @Override
+    public String returnAddressAndWorkSchedule() {
+        return findById(1L).getAddress() + " " + findById(1L).getWorkSchedule();
```
что за магические числа? почему не 100?

dmitry-bizin replied:
Зачем 2 раза запрос в БД делать? Надо сделать 1 запрос в БД, получить результат и далее работать с результатом. У вас приложение просто в 2 раза тормозит на пустом месте.

dmitry-bizin commented on src/main/java/com/example/petshelter/service/impl/CatShelterServiceImpl.java:
```diff
@@ -0,0 +1,102 @@
+
+    @Override
+    public String returnSecurityContacts() {
+        return findById(1L).getSecurityContacts();
```
что за магические числа? почему не 100?

dmitry-bizin commented on src/main/java/com/example/petshelter/service/impl/CatShelterServiceImpl.java:
```diff
@@ -0,0 +1,102 @@
+
+    @Override
+    public String returnSafetyRecommendations() {
+        return findById(1L).getSafetyRecommendations();
```
что за магические числа? почему не 100?

dmitry-bizin commented on src/main/java/com/example/petshelter/service/impl/CatShelterServiceImpl.java:
```diff
@@ -0,0 +1,102 @@
+        if(!validationService.validateString(name)) {
+            throw new ValidationException(catService.toString());
+        }
+        findById(1L).getCats().add(catService.findByName(name));
```
что за магические числа? почему не 100?

dmitry-bizin commented on src/main/java/com/example/petshelter/service/impl/CatShelterServiceImpl.java:
```diff
@@ -0,0 +1,102 @@
+        if(!validationService.validateString(phoneNumber)) {
+            throw new ValidationException(catOwnerService.toString());
+        }
+        findById(1L).getCatOwners().add(catOwnerService.findByPhoneNumber(phoneNumber));
```
что за магические числа? почему не 100?

dmitry-bizin commented on src/main/java/com/example/petshelter/service/impl/DogOwnerServiceImpl.java:
```diff
@@ -0,0 +1,75 @@
+    @Override
+    public DogOwner findById(Long id) {
+        if (dogOwnerRepository.findById(id).isPresent()) {
+            return dogOwnerRepository.findById(id).get();
```
Зачем 2 раза запрос в БД делать? Надо сделать 1 запрос в БД, получить результат и далее работать с результатом. У вас приложение просто в 2 раза тормозит на пустом месте.

dmitry-bizin commented on src/main/java/com/example/petshelter/service/impl/DogOwnerServiceImpl.java:
```diff
@@ -0,0 +1,75 @@
+    public DogOwner deleteById(Long id) {
+        if (dogOwnerRepository.findById(id).isPresent()) {
+            dogOwnerRepository.deleteById(id);
+            return dogOwnerRepository.findById(id).get();
```
Зачем 2 раза запрос в БД делать? Надо сделать 1 запрос в БД, получить результат и далее работать с результатом. У вас приложение просто в 2 раза тормозит на пустом месте.

dmitry-bizin replied:
и как это вы после удаления хотите из БД получить? это ко всем местам относится, у вас везде копипаст с одними и тем же ошибками

dmitry-bizin commented on src/main/java/com/example/petshelter/service/impl/DogServiceImpl.java:
```diff
@@ -0,0 +1,69 @@
+    @Override
+    public Dog findById(Long id) {
+        if (dogRepository.findById(id).isPresent()) {
+            return dogRepository.findById(id).get();
```
Зачем 2 раза запрос в БД делать? Надо сделать 1 запрос в БД, получить результат и далее работать с результатом. У вас приложение просто в 2 раза тормозит на пустом месте.

dmitry-bizin commented on src/main/java/com/example/petshelter/service/impl/DogServiceImpl.java:
```diff
@@ -0,0 +1,69 @@
+    public Dog deleteById(Long id) {
+        if (dogRepository.findById(id).isPresent()){
+            dogRepository.deleteById(id);
+            return dogRepository.findById(id).get();
```
Зачем 2 раза запрос в БД делать? Надо сделать 1 запрос в БД, получить результат и далее работать с результатом. У вас приложение просто в 2 раза тормозит на пустом месте.

dmitry-bizin commented on src/main/java/com/example/petshelter/service/impl/DogShelterServiceImpl.java:
```diff
@@ -0,0 +1,103 @@
+    @Override
+    public DogShelter findById(Long id) {
+        if (dogShelterRepository.findById(id).isPresent()) {
+            return dogShelterRepository.findById(id).get();
```
Зачем 2 раза запрос в БД делать? Надо сделать 1 запрос в БД, получить результат и далее работать с результатом. У вас приложение просто в 2 раза тормозит на пустом месте.

dmitry-bizin commented on src/main/java/com/example/petshelter/service/impl/DogShelterServiceImpl.java:
```diff
@@ -0,0 +1,103 @@
+    public DogShelter deleteById(Long id) {
+        if (dogShelterRepository.findById(id).isPresent()) {
+            dogShelterRepository.deleteById(id);
+            return dogShelterRepository.findById(id).get();
```
Зачем 2 раза запрос в БД делать? Надо сделать 1 запрос в БД, получить результат и далее работать с результатом. У вас приложение просто в 2 раза тормозит на пустом месте.

dmitry-bizin commented on src/main/java/com/example/petshelter/service/impl/MessageToVolunteerServiceImpl.java:
```diff
@@ -0,0 +1,55 @@
+    @Override
+    public MessageToVolunteer findById(Long id) {
+        if (messageToVolunteerRepository.findById(id).isPresent()) {
+            return messageToVolunteerRepository.findById(id).get();
```
Зачем 2 раза запрос в БД делать? Надо сделать 1 запрос в БД, получить результат и далее работать с результатом. У вас приложение просто в 2 раза тормозит на пустом месте.

dmitry-bizin commented on src/main/java/com/example/petshelter/service/impl/MessageToVolunteerServiceImpl.java:
```diff
@@ -0,0 +1,55 @@
+    public MessageToVolunteer deleteById(Long id) {
+        if (messageToVolunteerRepository.findById(id).isPresent()) {
+            messageToVolunteerRepository.deleteById(id);
+            return messageToVolunteerRepository.findById(id).get();
```
Зачем 2 раза запрос в БД делать? Надо сделать 1 запрос в БД, получить результат и далее работать с результатом. У вас приложение просто в 2 раза тормозит на пустом месте.

dmitry-bizin commented on src/main/java/com/example/petshelter/service/impl/ReportServiceImpl.java:
```diff
@@ -0,0 +1,52 @@
+    @Override
+    public Report findById(Long id) {
+        if (reportRepository.findById(id).isPresent()) {
+            return reportRepository.findById(id).get();
```
Зачем 2 раза запрос в БД делать? Надо сделать 1 запрос в БД, получить результат и далее работать с результатом. У вас приложение просто в 2 раза тормозит на пустом месте.

dmitry-bizin commented on src/main/java/com/example/petshelter/service/impl/ReportServiceImpl.java:
```diff
@@ -0,0 +1,52 @@
+    public Report deleteById(Long id) {
+        if (reportRepository.findById(id).isPresent()) {
+            reportRepository.deleteById(id);
+            return reportRepository.findById(id).get();
```
Зачем 2 раза запрос в БД делать? Надо сделать 1 запрос в БД, получить результат и далее работать с результатом. У вас приложение просто в 2 раза тормозит на пустом месте.

dmitry-bizin commented on src/main/java/com/example/petshelter/service/impl/TrialPeriodServiceImpl.java:
```diff
@@ -0,0 +1,56 @@
+    @Override
+    public TrialPeriod findById(Long id) {
+        if (trialPeriodRepository.findById(id).isPresent()){
+            return trialPeriodRepository.findById(id).get();
```
Зачем 2 раза запрос в БД делать? Надо сделать 1 запрос в БД, получить результат и далее работать с результатом. У вас приложение просто в 2 раза тормозит на пустом месте.

dmitry-bizin commented on src/main/java/com/example/petshelter/service/impl/TrialPeriodServiceImpl.java:
```diff
@@ -0,0 +1,56 @@
+    public TrialPeriod deleteById(Long id) {
+        if (trialPeriodRepository.findById(id).isPresent()){
+            trialPeriodRepository.deleteById(id);
+            return trialPeriodRepository.findById(id).get();
```
Зачем 2 раза запрос в БД делать? Надо сделать 1 запрос в БД, получить результат и далее работать с результатом. У вас приложение просто в 2 раза тормозит на пустом месте.

dmitry-bizin commented on src/main/java/com/example/petshelter/service/impl/UserServiceImpl.java:
```diff
@@ -0,0 +1,63 @@
+    @Override
+    public User findById(Long id) {
+        if (userRepository.findById(id).isPresent()) {
+            return userRepository.findById(id).get();
```
Зачем 2 раза запрос в БД делать? Надо сделать 1 запрос в БД, получить результат и далее работать с результатом. У вас приложение просто в 2 раза тормозит на пустом месте.

dmitry-bizin commented on src/main/java/com/example/petshelter/service/impl/UserServiceImpl.java:
```diff
@@ -0,0 +1,63 @@
+    public User deleteById(Long id) {
+        if (userRepository.findById(id).isPresent()) {
+            userRepository.deleteById(id);
+            return userRepository.findById(id).get();
```
Зачем 2 раза запрос в БД делать? Надо сделать 1 запрос в БД, получить результат и далее работать с результатом. У вас приложение просто в 2 раза тормозит на пустом месте.

dmitry-bizin commented on src/main/java/com/example/petshelter/service/impl/DogShelterServiceImpl.java:
```diff
@@ -0,0 +1,103 @@
+
+    @Override
+    public String returnInformation() {
+        return findById(1L).getInformation();
```
что за магические числа? почему не 100?

dmitry-bizin commented on src/main/java/com/example/petshelter/service/impl/DogShelterServiceImpl.java:
```diff
@@ -0,0 +1,103 @@
+
+    @Override
+    public String returnAddressAndWorkSchedule() {
+        return findById(1L).getAddress() + " " + findById(1L).getWorkSchedule();
```
что за магические числа? почему не 100?

dmitry-bizin commented on src/main/java/com/example/petshelter/service/impl/DogShelterServiceImpl.java:
```diff
@@ -0,0 +1,103 @@
+
+    @Override
+    public String returnSecurityContacts() {
+        return findById(1L).getSecurityContacts();
```
что за магические числа? почему не 100?

dmitry-bizin commented on src/main/java/com/example/petshelter/service/impl/DogShelterServiceImpl.java:
```diff
@@ -0,0 +1,103 @@
+
+    @Override
+    public String returnSafetyRecommendations() {
+        return findById(1L).getSafetyRecommendations();
```
что за магические числа? почему не 100?

dmitry-bizin commented on src/main/java/com/example/petshelter/service/impl/DogShelterServiceImpl.java:
```diff
@@ -0,0 +1,103 @@
+        if(!validationService.validateString(name)) {
+            throw new ValidationException(dogService.toString());
+        }
+        findById(1L).getDogs().add(dogService.findByName(name));
```
что за магические числа? почему не 100?

dmitry-bizin commented on src/main/java/com/example/petshelter/service/impl/DogShelterServiceImpl.java:
```diff
@@ -0,0 +1,103 @@
+        if(!validationService.validateString(phoneNumber)) {
+            throw new ValidationException(dogOwnerService.toString());
+        }
+        findById(1L).getDogOwners().add(dogOwnerService.findByPhoneNumber(phoneNumber));
```
что за магические числа? почему не 100?

SergeyGolybtsov committed:
Доработал тесты.

MarselMingaripov committed:
исправил методы поиск по ид и удаление по ид

elenamatvienko committed:
Доработала контроллеры CatShelter, DogShelter.  Доработала описание API. Перенесла JavaDoc в интерфейсы сервисов.

elenamatvienko committed:
Merge remote-tracking branch 'origin/dev' into dev

# Conflicts:
#	src/main/java/com/example/petshelter/service/CatService.java
#	src/main/java/com/example/petshelter/service/impl/CatServiceImpl.java

MarselMingaripov committed:
добавил бронь собак на сутки, добавил методы замены статусов животных и поиск по статусам в контроллеры собак и кошек

MarselMingaripov committed:
Merge branch 'dev' into marselFeature-1

# Conflicts:
#	src/main/java/com/example/petshelter/controller/DogController.java
#	src/main/java/com/example/petshelter/service/DogService.java
#	src/main/java/com/example/petshelter/service/impl/CatServiceImpl.java

MarselMingaripov committed:
добавил метод отдать животное на испытательный срок

elenamatvienko committed:
Доработала CatShelterService, CatShelterServiceImpl, CatShelterController, DogShelterService, DogShelterServiceImpl, DogShelterController (заменила в инф-ых методах 1L на id).  Доработала ValidationService и ValidationServiceImpl (добавила валидацию messageToVolunteer, report, trialPeriod, user)

vladimirivn committed:
Добавил тесты по UserService, CatOwnerService, CatShelterService

vladimirivn committed:
Merge remote-tracking branch 'origin/dev' into dev

vladimirivn committed:
исправил тест в UserServiceTest

vladimirivn committed:
Merge branch 'vladimirivnFeature-1' into dev

MarselMingaripov committed:
добавил HelloWorld бота

MarselMingaripov committed:
добавил конфиги, простую клавиатуру, контроллер, настроил лиссенер

vladimirivn committed:
Добавил новые тесты

elenamatvienko committed:
Написала тесты MessageToVolunteer, TrialPeriod, Report

MarselMingaripov committed:
настроил меню бота, добавил инициализацию приютов

MarselMingaripov committed:
Merge branch 'marselFeature-1' into dev

vladimirivn committed:
Merge branch 'dev' into vladimirivnFeature-1

vladimirivn committed:
Добавил новые тесты

vladimirivn committed:
Добавил новые тесты, исправил ошибки

elenamatvienko committed:
Написала методы addCatConsult, addDogConsult в сервисах(Impl) и контроллерах

elenamatvienko committed:
Merge remote-tracking branch 'origin/dev' into dev

vladimirivn committed:
Добавил новые тесты

vladimirivn committed:
Merge branch 'vladimirivnFeature-1' into dev

vladimirivn committed:
Добавил новые тесты, исправил ошибки

MarselMingaripov committed:
автоматическая отправка оповещений о событиях волонтерам. регистрация волонтеров

vladimirivn committed:
Добавил тест ValidationServiceTest, исправил ошибки

MarselMingaripov committed:
добавил тесты на метод создать кота в контролеере котов

MarselMingaripov committed:
Merge branch 'dev' into marselFeature-1

# Conflicts:
#	src/test/java/com/example/petshelter/controller/CatControllerTest.java

MarselMingaripov committed:
добавил конструктор на пользователя

dmitry-bizin commented on Bot/src/main/java/com/example/bot/controller/BotController.java:
```diff
@@ -0,0 +1,60 @@
+        Map<String, Long> uriVariables = new HashMap<>();
+        uriVariables.put("id", 1L);
+        ResponseEntity<String> responseEntity = new RestTemplate().getForEntity(
+                "http://localhost:8080/catShelter/information/{id}", String.class, uriVariables);
```
так, а что это такое? для чего это вообще?

dmitry-bizin replied:
не нужно такое делать, зачем?

dmitry-bizin commented on Bot/src/main/java/com/example/bot/listener/TelegramBotUpdatesListener.java:
```diff
@@ -0,0 +1,203 @@
+                        CallbackQuery callbackQuery = update.callbackQuery();
+                        String data = callbackQuery.data();
+                        switch (data) {
+                            case "/dogs":
```
строки надо выносить в константы

SergeyGolybtsov committed:
Merge branch 'dev' into SergeyFicha

MarselMingaripov committed:
удалил модуль бота, перенес все в монолит. закончил 2 этап меню в боте в приюте кошек

SergeyGolybtsov committed:
Доработал тесты 2.

vladimirivn committed:
Тест CatOwnerControllerTest - с ошибкой

vladimirivn committed:
Тест CatOwnerControllerTest - с ошибкой

vladimirivn committed:
Тест CatOwnerControllerTest, проверка shouldReturn200WhenCreateCorrectFieldsCatOwner() - исправил ошибку

elenamatvienko committed:
Начала писать тесты на контроллеры

vladimirivn committed:
Дописал тесты в CatShelterServiceTest,
UserServiceTest - 100%,

vladimirivn committed:
Тест CatOwnerControllerTest, проверка

MarselMingaripov committed:
настроил принятие отчета, сохранение в бд, отправку уведомлений волонтерам

vladimirivn committed:
Дописал тесты в CatOwnerControllerTest

vladimirivn committed:
CatOwnerControllerTest, добавил проверку

vladimirivn committed:
CatOwnerControllerTest - 87% покрытие

elenamatvienko committed:
Впроцессе написания тестов на контроллеры

elenamatvienko committed:
Merge remote-tracking branch 'origin/dev' into dev

# Conflicts:
#	src/test/java/com/example/petshelter/controller/TrialPeriodControllerTest.java

vladimirivn committed:
добавил тест для DogOwnerControllerTest

elenamatvienko committed:
Merge remote-tracking branch 'origin/dev' into dev

elenamatvienko committed:
Написала CatControllerTest, DogControllerTest покрытие 100%

elenamatvienko committed:
Merge branch 'elenaFeature1' into dev

vladimirivn committed:
Доделал CatOwnerControllerTest,DogOwnerControllerTest. Начал делать CatShelterControllerTest

MarselMingaripov committed:
настроил таймер проверки ежедневных отчетов, оповещения для пользователя и волонтера

MarselMingaripov committed:
отправка отчета пользователя записывается в его испытательный срок

elenamatvienko committed:
Написала MessageToVolunteerControllerTest покрытие 100%

elenamatvienko committed:
Merge branch 'elenaFeature1' into dev

elenamatvienko committed:
Merge remote-tracking branch 'origin/dev' into dev

elenamatvienko committed:
Написала ReportControllerTest покрытие 100%

elenamatvienko committed:
Merge branch 'elenaFeature1' into dev

# Conflicts:
#	src/main/java/com/example/petshelter/entity/Report.java

MarselMingaripov committed:
добавил всю логику бота для собак

MarselMingaripov committed:
начал писать тесты на бота

MarselMingaripov committed:
начал писать тесты на бота, покрыл 60%

vladimirivn committed:
CatOwnerControllerTest, CatShelterControllerTest - 100% покрытие
DogOwnerControllerTest - 100% без метода Передача животного усыновителю, установка испытательного срока

vladimirivn committed:
UserControllerTest - 100% покрытие

vladimirivn committed:
Дополнение тестов для CatOwnerServiceTest и CatShelterServiceTest

elenamatvienko committed:
Написала TrialPeriodControllerTest, TrialPeriodServiceTest покрытие 100%, ReportServiceTest покрытие 85%, MessageToVolunteerServiceTest покрытие 77%

elenamatvienko committed:
Merge remote-tracking branch 'origin/dev' into dev

vladimirivn committed:
Добавил тесты в CatShelterServiceTest

vladimirivn committed:
CatShelterServiceTest -92%

vladimirivn committed:
Добавил DogOwnerServiceTest, DogShelterControllerTest, DogShelterServiceTest

MarselMingaripov committed:
Merge branch 'dev' into marselFeature-1

MarselMingaripov committed:
поправил тесты немножко

SergeyGolybtsov committed:
Доработал тесты 4.

dmitry-bizin commented on src/main/java/com/example/petshelter/bot/listener/TelegramBotUpdatesListener.java:
```diff
@@ -0,0 +1,364 @@
+                        CallbackQuery callbackQuery = update.callbackQuery();
+                        String data = callbackQuery.data();
+                        switch (data) {
+                            case "/dogs":
```
команды надо выносить в константы, чтобы потом использовать везде, а не дублировать при отправке и здесь в обработке

dmitry-bizin commented on src/main/java/com/example/petshelter/bot/listener/TelegramBotUpdatesListener.java:
```diff
@@ -0,0 +1,364 @@
+                        logger.info(data);
+                    }
+                    return;
+                } catch (NullPointerException e) {
```
очень плохая практика - ловить NPE, это просто игнорирование багов. Баги надо исправлять, а не игнорировать

dmitry-bizin commented on src/main/java/com/example/petshelter/bot/listener/TelegramBotUpdatesListener.java:
```diff
@@ -0,0 +1,364 @@
+                                        .getPhoneNumber() + " он отправил отчет! Посмотрите, пожалуйста, что он там наприсылал"));
+                                sendMessageInCatShelterMenu(chatId, "Спасибо!");
+                            }
+                        } catch (ArrayIndexOutOfBoundsException e){
```
аналогично - надо исправлять такое исключение, а не игнорировать

SergeyGolybtsov committed:
Merge remote-tracking branch 'origin/dev' into dev

MarselMingaripov committed:
вынес строки в константы в TelegramBotUpdatesListener

MarselMingaripov committed:
Merge branch 'dev' into marselFeature-1

vladimirivn committed:
Исправил ошибки в DogOwnerController,DogShelterController. Исправил DogOwnerControllerTest.Добавил тесты в DogShelterControllerTest -100%

vladimirivn committed:
Merge remote-tracking branch 'origin/dev' into dev

vladimirivn committed:
Подправил сервисы

vladimirivn committed:
Подправил в CatShelterController и DogShelterController описания операций для Swagger-а
